examples/step-64/step-64.cu

```diff
@@ -0,0 +1,576 @@
+                                             constraints);
+    constraints.close();
+    system_matrix_dev.reset(
+      new HelmholtzOperator<dim, fe_degree>(dof_handler, constraints));
+
+    ghost_solution_host.reinit(locally_owned_dofs,
+                               locally_relevant_dofs,
+                               mpi_communicator);
+    solution_dev.reinit(locally_owned_dofs, mpi_communicator);
+    system_rhs_dev.reinit(locally_owned_dofs, mpi_communicator);
+  }
+
+
+
+  template <int dim, int fe_degree>
+  void HelmholtzProblem<dim, fe_degree>::assemble_rhs()
+  {
+    LinearAlgebra::distributed::Vector<double, MemorySpace::Host>
+                      system_rhs_host(locally_owned_dofs,
+                      locally_relevant_dofs,
+                      mpi_communicator);
+    const QGauss<dim> quadrature_formula(fe_degree + 1);
+
+    FEValues<dim> fe_values(fe,
+                            quadrature_formula,
+                            update_values | update_quadrature_points |
+                              update_JxW_values);
+
+    const unsigned int dofs_per_cell = fe.dofs_per_cell;
+    const unsigned int n_q_points    = quadrature_formula.size();
+
+    Vector<double> cell_rhs(dofs_per_cell);
+
+    std::vector<types::global_dof_index> local_dof_indices(dofs_per_cell);
+
+    for (const auto &cell : dof_handler.active_cell_iterators())
+      if (cell->is_locally_owned())
+        {
+          cell_rhs = 0;
+
+          fe_values.reinit(cell);
+
+          for (unsigned int q_index = 0; q_index < n_q_points; ++q_index)
+            {
+              for (unsigned int i = 0; i < dofs_per_cell; ++i)
+                cell_rhs(i) += (fe_values.shape_value(i, q_index) * 1.0 *
+                                fe_values.JxW(q_index));
+            }
+
+          // Finally, transfer the contributions from @p cell_rhs into the global
+          // objects. Set the constraints to zero. This is necessary for CG to
+          // converge since the ansatz and solution space have these degrees of
+          // freedom constrained as well.
+          // The other solution is modifying vmult() so that the source
+          // vector sets the contrained dof to zero.
+          cell->get_dof_indices(local_dof_indices);
+          constraints.distribute_local_to_global(cell_rhs,
+                                                 local_dof_indices,
+                                                 system_rhs_host);
+        }
+    system_rhs_host.compress(VectorOperation::add);
+
+    // We can't directly copy the values from the host to the device but need to
+    // use an intermediate object of type LinearAlgebra::ReadWriteVector to
+    // construct the correct communication pattern.
+    LinearAlgebra::ReadWriteVector<double> rw_vector(locally_owned_dofs);
+    rw_vector.import(system_rhs_host, VectorOperation::insert);
+    system_rhs_dev.import(rw_vector, VectorOperation::insert);
```

Comment:
I guess it is on our todo list to make this import also work for `LA::dist::Vector<Number,Host>`? But that can definitely wait until after the release.

Reply:
Yes, we should do that for the next release.